L2 support

snreynolds opened the pull request.

willpote commented on README.md:
````diff
@@ -22,6 +22,18 @@ First create a `.env` file in the root of the project and configure:
 
+To run on chains other than mainnet set up a connection by specifying the environment variable
+
+```JSON_RPC_PROVIDER_{CHAIN} = '<JSON_RPC_PROVIDER>'
````
Spacing

willpote reviewed (commented):
Looking good so far!

Before merging would be great if we could manually test on all networks. 

I think right now most networks fail to get V3 quotes because of the QuoteProvider not having the QuoterV2 contract [here](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/v3/quote-provider.ts#L17)

Good news is Noah just deployed it to all networks at 0x61fFE014bA17989E743c5F6cB21bF9697530B21e so we can update it now

Also since Polygon is now looking possible, think we need to add support for it

willpote commented:
Also realized that for getting gas prices on some networks, we can't use the EIP1559 gas price provider, e.g. Optimism doesn't support it. I added a new gas price provider for legacy networks and plugged it into the CLI. I was able to get a quote for Optimism!

```
./bin/cli quote --tokenIn 0x7F5c764cBc14f9669B88837ca1490cCa17c31607 --tokenOut 0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1 --amount 200000 --exactIn --minSplits 1 --router alpha --chainId 10 --debug
```

willpote commented on src/routers/alpha-router/alpha-router.ts:
```diff
@@ -492,6 +492,16 @@ export class AlphaRouter
+    ) {
+      log.error(`This chain ${this.chainId} does not support v2 trades`);
+      return null;
+    }
```
Done this one

snreynolds replied:
yep, makes sense

snreynolds commented:
Ok couple of minor bug fixes and edits since the last review:

- fix:  [catch](https://github.com/Uniswap/smart-order-router/pull/39/commits/fd3fc76c436ec42eb51946c8a6cd70ec4a85761a#diff-e189334c133d80074b95d2b9d8da279bdb3d62e55d6f48694a436642f3024f5fR280) the v3SubgraphProvider error if there’s no subgraph url. This was erroring bc we initialized it in `V3SubgraphProviderWithFallBacks` even if the network doesn't have a dedicated url.
- fix: infura has trouble handling Koran eip1559 requests so removed it from eip1559 [supported chains list](https://github.com/Uniswap/smart-order-router/pull/39/commits/be575dbf01c66e8a0896fc4f66d3afb571127359#diff-6a9cea9971544474cf3aec0fd1298469edc36810041062758194c9ed6d3373f0R11) 
- added cli commands for all supported networks in the README for easy future testing
- added more tokens for each chains static subgraph
- created fallback [batchParams](https://github.com/Uniswap/smart-order-router/pull/39/commits/c40630eaf1cca72fbdcdd5c1e279ff73e73c2237#diff-25b82bb69240d110ebc90a2530124d622e964584fec4821a85dc2f1570a334dbR173) for Arbitrum support

willpote reviewed (approved):
Great job! Couple nits, but feel free to merge
